salt/hosts/init.sls

```diff
@@ -1,28 +1,13 @@
-{%- set minealias    = salt['pillar.get']('hostsfile:alias', 'network.ip_addrs') %}
-{%- set minions      = salt['pillar.get']('hostsfile:minions', '*') %}
-{%- set pillar_hosts = salt['pillar.get']('hostsfile:hosts', {}) %}
-{%- set mine_hosts   = salt['mine.get'](minions, minealias) %}
-
-{%- set hosts = {} %}
-{%- if mine_hosts is defined %}
-{%-   do hosts.update(mine_hosts) %}
-{%- endif %}
-{%- do hosts.update(pillar_hosts) %}
-
-{%- for fqdn, addrlist in hosts.items() %}
-{{ fqdn }}-host-entry:
+# Internally we refer to hosts by their machine ID, which is also used as the minion ID.
+{%- for minion_id, addrlist in salt['mine.get']('*', 'network.ip_addrs').items() %}
+{{ minion_id }}-host-entry:
   host.present:
 {% if addrlist is string %}
     - ip: {{ addrlist }}
 {% else %}
     - ip: {{ addrlist|first }}
 {% endif %}
```

Comment:
Because:

1. we want to use `api.infra.cluster.local` for referring to the API server, so we could move towards solving [this issue](https://trello.com/c/yCbzToby)
2. we will need to react to IP changes for solving [this](https://trello.com/c/a8dfFSkP), and we will need to use two different states unless we want massive updates...
3. it is not such a big change :-)